index/index.go

```diff
@@ -19,3 +25,15 @@ type Identifier struct {
 type FileList struct {
 	FileTypes map[string]bool
 }
+
+// BuildIndex : Returns an initialised index of the type at the config key.
+func BuildIndex(indexConfigKey string) Index {
+
+	// Initialises and returns an instance of InMemory
+	if strings.Compare(viper.GetString(indexConfigKey+".type"), "in-memory") == 0 {
+		return newInMemory()
+	}
+
+	// If index configuration is not complete or cannot be instantiated, return nil
+	return nil
```

Comment:
Do we not want to log in a warning in this case? Or add handling of this case wherever it's called?